### PR TITLE
Setup tests

### PR DESCRIPTION
**Pull request by arunkannawadi**

### CODE REVIEW

**Comment by arunkannawadi** [comment]
@esheldon I'm not quite ready to hand this off to you yet. It's a collaboration meeting week, and S3DF is down, so I can't work on this right now. I can hand it off to you next week.

**Comment by esheldon** [comment]
What do you mean by "hand off"?

I'm mainly going to be adding tests to make sure the task is giving results as expected.  The most detailed tests will still be in the `descwl_coadd` package.

**Comment by arunkannawadi** [comment]
This is ready for a review. I had originally intended to set these up and merge, and for you to add further tests on top of these on another branch. I think that might still be for the best here.

File: descwl_coadd_task/tests/test_warps.py
Comment (arunkannawadi) [inline descwl_coadd_task/tests/test_warps.py:48]: We can speed the tests further by filling the pixel values once in `setUpClass` and setting mask bits in the tests. We will still have to clear the masks in the `tearDown` method which gets run after every test.

File: descwl_coadd_task/tests/test_warps.py
Comment (arunkannawadi) [inline descwl_coadd_task/tests/test_warps.py:29]: These values are a bit high for unit tests, but mocking up an appropriate geometry that is sufficiently non-trivial was taking up time and I decided to base this off of a DC2 image.

**Comment by esheldon** [comment]
Arun is this ready for review?

**Comment by arunkannawadi** [comment]
Yes, Erin, this PR is ready for a review.

File: .github/workflows/tests.yaml
Comment (beckermr) [inline .github/workflows/tests.yaml:36]:
```suggestion
```

This is deprecated. https://conda-forge.org/news/2024/07/29/sunsetting-mambaforge/

File: .github/workflows/tests.yaml
Comment (beckermr) [inline .github/workflows/tests.yaml:42]:
```suggestion
          conda install -q stackvana=0
```

conda uses the mamba solver now.

File: setup.cfg
Comment (beckermr) [inline setup.cfg:11]: Why was this stuff removed?

File: descwl_coadd_task/tests/test_warps.py
Comment (beckermr) [inline descwl_coadd_task/tests/test_warps.py:356]: What does this do?

**Review by beckermr — CHANGES_REQUESTED** [review]
Some initial comments from me.

File: descwl_coadd_task/tests/test_warps.py
Comment (esheldon) [inline descwl_coadd_task/tests/test_warps.py:36]: Don't the data get modified in place?  In that case don't we want to remake the data for each unit test?

File: descwl_coadd_task/tests/test_warps.py
Comment (esheldon) [reply descwl_coadd_task/tests/test_warps.py:29]: In my commits (since discarded) I was able to use smaller images. Should I re-push those changes?

**Comment by esheldon** [comment]
Arun our reviews are complete

**Comment by arunkannawadi** [comment]
Thanks Ern and Matt. I'll address the review comments next week when I'm back from vacation.

File: descwl_coadd_task/tests/test_warps.py
Comment (arunkannawadi) [reply descwl_coadd_task/tests/test_warps.py:29]: This has to be adjusted together with a few other changes in `_make_data`. Is it a question of simulation time to have larger images?

File: descwl_coadd_task/tests/test_warps.py
Comment (arunkannawadi) [reply descwl_coadd_task/tests/test_warps.py:356]: It checks for memory/resource leaks. Just a trivial subclassing of it is sufficient for the testing framework to pick it up and check for leaks.

File: descwl_coadd_task/tests/test_warps.py
Comment (arunkannawadi) [reply descwl_coadd_task/tests/test_warps.py:36]: This doesn't set the pixel values. It just sets up the geometry, which can be the same for all the tests.

**Comment by esheldon** [comment]
Shall we merge this and then I can add more tests in a separate PR?

**Review by esheldon — APPROVED** [review]

File: setup.cfg
Comment (arunkannawadi) [reply setup.cfg]: I don't think this is recommended or supported any more? There is a `flake8` config above this with the same rules anyway.

**Comment by arunkannawadi** [comment]
Should we wait for Matt to approve or at least dismiss the change request? I _could_ bypass the branch protection and merge, but I'd rather not.

File: descwl_coadd_task/tests/test_warps.py
Comment (esheldon) [reply descwl_coadd_task/tests/test_warps.py:29]: Yes time/space etc.  I'll worry about that in other PRs

File: descwl_coadd_task/tests/test_warps.py
Comment (esheldon) [reply descwl_coadd_task/tests/test_warps.py:48]: Its simpler and less error prone to remake the data each time rather than deal with what is effectively a global state paradigm.  I can look into it in a followup PR

**Comment by esheldon** [comment]
Let's wait for Matt who is back from vacation tomorrow

File: descwl_coadd_task/tests/test_warps.py
Comment (arunkannawadi) [reply descwl_coadd_task/tests/test_warps.py:48]: Just to be clear, `setUp` gets run before every test case. It's not global.

Comment (esheldon) [reply descwl_coadd_task/tests/test_warps.py:48]: I see.  I don't use this way of doing unit tests, I  just write functions `test_*` that get automatically run by `pytest`

How can you do different versions of setUp for different tests, do you need a new class?

What's the advantage of `setUp` versus having a function that creates the data, in which case you can send keywords to change the data.

**Review by beckermr — APPROVED** [review]

**Pull request merged**
